src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -1,14 +1,177 @@
+
+
+    private PersistenceService persistenceService;
+    private ParseRequestService parseRequestService;
+    private ValidationModelService validationModelService;
+    RequestEntity requestEntity;
+    AssignmentRequest assignmentRequest;
+
+    public DeleteRoleAssignmentOrchestrator(PersistenceService persistenceService,
+                                            ParseRequestService parseRequestService,
+                                            ValidationModelService validationModelService) {
+        this.persistenceService = persistenceService;
+        this.parseRequestService = parseRequestService;
+        this.validationModelService = validationModelService;
+    }
+
+    public ResponseEntity<Object> deleteRoleAssignment(String actorId,
+                                                       String process,
+                                                       String reference) throws Exception {
+        List<RequestedRole> requestedRoles = null;
+
+        //1. create the request Object
+        Request request = parseRequestService.prepareRequestObject(process, reference);
+        assignmentRequest = new AssignmentRequest();
+
+        //2. Call persistence service to store only the request
+        requestEntity = persistenceService.persistRequest(request);
+        requestEntity.setHistoryEntities(new HashSet<>());
+        request.setId(requestEntity.getId());
+
+        //3. retrieve all assignment records based on actorId/process+reference
+        if (actorId != null) {
+            requestedRoles = persistenceService.getRoleAssignment(UUID.fromString(actorId));
+
+        } else if (process != null && reference != null) {
+            requestedRoles = persistenceService.getExistingRoleByProcessAndReference(
+                process,
+                reference,
+                Status.LIVE.toString()
+            );
+
+        }
+
+        //4. call validation rule
+        if (requestedRoles != null && !requestedRoles.isEmpty()) {
+            validationByDrool(request, requestedRoles);
+        }
+
+        //5. persist the  requested roles  and update status
+        updateStatusAndPersist(request);
+
+        //6. check status updated by drools and take decision
+        checkAllDeleteApproved(assignmentRequest);
+
+
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+
+    }
+
+    private void validationByDrool(Request request, List<RequestedRole> requestedRoles) throws Exception {
+        assignmentRequest.setRequest(request);
+        assignmentRequest.setRequestedRoles(requestedRoles);
+
+        //calling drools rules for validation
+        validationModelService.validateRequest(assignmentRequest);
+
+
+    }
+
+    private void updateStatusAndPersist(Request request) {
+        for (RequestedRole requestedRole : assignmentRequest.getRequestedRoles()) {
+            requestedRole.setRequest(request);
+            if (!requestedRole.getStatus().equals(Status.APPROVED)) {
```

Comment:
we need to set DELETE_APPROVED directly in Drools, if it is not DELETE_APPROVED then set it to DELETE_REJECTED

Reply:
Done.